panic when there are 10K volumes in the cluster

When there are 10K volumes in the cluster, I found `csi-provisioner` v2.1.0 would hit nil pointer panic frequently, someone has clue why hit `unrecognized notification: <nil>` issue in `csi-provisioner` v2.1.0 ? thanks.

```
0311 07:58:11.639403       1 feature_gate.go:235] Setting GA feature gate Topology=true. It will be removed in a future release.
I0311 07:58:11.639643       1 feature_gate.go:243] feature gates: &{map[Topology:true]}
I0311 07:58:11.639665       1 csi-provisioner.go:132] Version: v2.1.0-0-gb7a8fe9fe
I0311 07:58:11.639713       1 csi-provisioner.go:155] Building kube configs for running in cluster...
I0311 07:58:11.914028       1 connection.go:153] Connecting to unix:///csi/csi.sock
I0311 07:58:11.914618       1 common.go:111] Probing CSI driver for readiness
I0311 07:58:11.916555       1 csi-provisioner.go:202] Detected CSI driver disk.csi.azure.com
I0311 07:58:11.932324       1 csi-provisioner.go:241] CSI driver supports PUBLISH_UNPUBLISH_VOLUME, watching VolumeAttachments
I0311 07:58:11.932522       1 csi-provisioner.go:337] Supports migration from in-tree plugin: kubernetes.io/azure-disk
I0311 07:58:11.933012       1 controller.go:753] Using saving PVs to API server in background
I0311 07:58:11.936210       1 leaderelection.go:243] attempting to acquire leader lease kube-system/disk-csi-azure-com...
I0311 07:58:11.946816       1 leaderelection.go:253] successfully acquired lease kube-system/disk-csi-azure-com
I0311 07:58:11.947096       1 leader_election.go:205] became leader, starting
I0311 07:58:11.947392       1 reflector.go:219] Starting reflector *v1.StorageClass (1h0m0s) from k8s.io/client-go/informers/factory.go:134
I0311 07:58:11.947769       1 reflector.go:219] Starting reflector *v1.PersistentVolumeClaim (15m0s) from k8s.io/client-go/informers/factory.go:134
I0311 07:58:11.947998       1 reflector.go:219] Starting reflector *v1.VolumeAttachment (1h0m0s) from k8s.io/client-go/informers/factory.go:134
I0311 07:58:11.948335       1 reflector.go:219] Starting reflector *v1.CSINode (1h0m0s) from k8s.io/client-go/informers/factory.go:134
I0311 07:58:11.948634       1 reflector.go:219] Starting reflector *v1.Node (1h0m0s) from k8s.io/client-go/informers/factory.go:134
I0311 07:58:17.431211       1 controller.go:838] Starting provisioner controller disk.csi.azure.com_aks-test-65251830-vmss000000_5d637f3e-7cab-433b-bd22-74899fd73436!
I0311 07:58:17.431991       1 clone_controller.go:66] Starting CloningProtection controller
I0311 07:58:17.432371       1 reflector.go:219] Starting reflector *v1.StorageClass (15m0s) from sigs.k8s.io/sig-storage-lib-external-provisioner/v6/controller/controller.go:875
I0311 07:58:17.432703       1 volume_store.go:97] Starting save volume queue
I0311 07:58:17.432954       1 reflector.go:219] Starting reflector *v1.PersistentVolume (15m0s) from sigs.k8s.io/sig-storage-lib-external-provisioner/v6/controller/controller.go:872
I0311 07:58:17.534286       1 clone_controller.go:84] Started CloningProtection controller
I0311 07:58:17.534456       1 reflector.go:219] Starting reflector *v1.PersistentVolumeClaim (15m0s) from github.com/kubernetes-csi/external-provisioner/pkg/controller/clone_controller.go:82
I0311 07:58:29.636634       1 trace.go:205] Trace[1106410694]: "Reflector ListAndWatch" name:github.com/kubernetes-csi/external-provisioner/pkg/controller/clone_controller.go:82 (11-Mar-2021 07:58:17.534) (total time: 12102ms):
Trace[1106410694]: ---"Objects listed" 11702ms (07:58:00.236)
Trace[1106410694]: [12.102129397s] [12.102129397s] END
I0311 07:58:29.932296       1 trace.go:205] Trace[460128162]: "Reflector ListAndWatch" name:sigs.k8s.io/sig-storage-lib-external-provisioner/v6/controller/controller.go:872 (11-Mar-2021 07:58:17.432) (total time: 12499ms):
Trace[460128162]: ---"Objects listed" 12305ms (07:58:00.738)
Trace[460128162]: [12.499283483s] [12.499283483s] END
I0311 07:58:31.432575       1 controller.go:887] Started provisioner controller disk.csi.azure.com_aks-test-65251830-vmss000000_5d637f3e-7cab-433b-bd22-74899fd73436!
E0311 08:13:17.137876       1 shared_informer.go:781] unrecognized notification: <nil>
E0311 08:13:17.233754       1 shared_informer.go:781] unrecognized notification: <nil>
E0311 08:13:30.131115       1 shared_informer.go:781] unrecognized notification: <nil>
E0311 08:28:17.534018       1 shared_informer.go:781] unrecognized notification: <nil>

E0311 08:43:30.236643       1 shared_informer.go:781] unrecognized notification: <nil>
E0311 08:58:17.140019       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 81 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1775c20, 0x25f89d0)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x1775c20, 0x25f89d0)
        /usr/lib/go-1.15/src/runtime/panic.go:969 +0x175
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/client-go/tools/cache/shared_informer.go:774 +0x194
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0005a3760)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0005b9f60, 0x1b660a0, 0xc000590a50, 0x1741401, 0xc000089680)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005a3760, 0x3b9aca00, 0x0, 0x1, 0xc000089680)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(...)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000355a80)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/client-go/tools/cache/shared_informer.go:771 +0x95
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc00015c5a0, 0xc00008bcd0)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x51
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x65
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1411d94]

goroutine 81 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x1775c20, 0x25f89d0)
        /usr/lib/go-1.15/src/runtime/panic.go:969 +0x175
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/client-go/tools/cache/shared_informer.go:774 +0x194
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0005a3760)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0005b9f60, 0x1b660a0, 0xc000590a50, 0x1741401, 0xc000089680)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005a3760, 0x3b9aca00, 0x0, 0x1, 0xc000089680)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(...)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000355a80)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/client-go/tools/cache/shared_informer.go:771 +0x95
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc00015c5a0, 0xc00008bcd0)
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x51
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        /home/azureuser/agent/_work/49/s/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x65
```

--- a/pkg/controller/clone_controller.go
+++ b/pkg/controller/clone_controller.go
@@ -79,8 +79,6 @@ func (p *CloningProtectionController) Run(ctx context.Context, threadiness int) 
 		}, time.Second, ctx.Done())
 	}
 
-	go p.claimInformer.Run(ctx.Done())
-
 	klog.Infof("Started CloningProtection controller")
 	<-ctx.Done()
 	klog.Info("Shutting down CloningProtection controller")

